sdk/resourcemanager/azure-resourcemanager-network/src/test/java/com/azure/resourcemanager/network/NetworkInterfaceOperationsTests.java

```diff
@@ -509,6 +511,197 @@ public void canAssociateNatGateway() {
+            .withAddressSpace("10.0.0.0/28")
+            .withSubnet(subnetName, "10.0.0.0/28")
+            .create();
+
+        NetworkInterface nic = networkManager.networkInterfaces()
+            .define(generateRandomResourceName("nic-", 15))
+            .withRegion(Region.US_EAST)
+            .withExistingResourceGroup(rgName)
+            .withExistingPrimaryNetwork(vnet)
+            .withSubnet(subnetName)
+            .withPrimaryPrivateIPAddressDynamic()
+            .withNewPrimaryPublicIPAddress()
+            .withPrimaryPublicIPAddressDeleteOptions(DeleteOptions.DELETE)
+            .create();
+
+        nic.refresh();
+        nic.update()
+            .defineSecondaryIPConfiguration("secondary")
+            .withExistingNetwork(vnet)
+            .withSubnet(subnetName)
+            .withPrivateIpAddressDynamic()
+            .withNewPublicIpAddress()
+            .withPublicIPAddressDeleteOptions(DeleteOptions.DELETE)
+            .attach()
+            .apply();
+        nic.refresh();
+        List<NetworkInterfaceIpConfigurationInner> ipConfigurationInnerList = nic.innerModel().ipConfigurations()
+            .stream().filter(ipConfigurationInner -> DeleteOptions.DELETE.equals(ipConfigurationInner.publicIpAddress().deleteOption()))
+            .collect(Collectors.toList());
+        Assertions.assertEquals(2, ipConfigurationInnerList.size());
+    }
+
+    @Test
+    public void canUpdateSpecifiedIpConfigurationWithDeleteOption() {
+        String subnetName = generateRandomResourceName("subnet-", 15);
+        resourceManager.resourceGroups().define(rgName).withRegion(Region.US_EAST).create();
+        Network vnet = networkManager.networks()
+            .define(generateRandomResourceName("vnet-", 15))
+            .withRegion(Region.US_EAST)
+            .withExistingResourceGroup(rgName)
+            .withAddressSpace("10.0.0.0/28")
+            .withSubnet(subnetName, "10.0.0.0/28")
+            .create();
+
+        NetworkInterface nic = networkManager.networkInterfaces()
+            .define(generateRandomResourceName("nic-", 15))
+            .withRegion(Region.US_EAST)
+            .withExistingResourceGroup(rgName)
+            .withExistingPrimaryNetwork(vnet)
+            .withSubnet(subnetName)
+            .withPrimaryPrivateIPAddressDynamic()
+            .withNewPrimaryPublicIPAddress()
+            .withPrimaryPublicIPAddressDeleteOptions(DeleteOptions.DELETE)
+            .defineSecondaryIPConfiguration("secondary")
+            .withExistingNetwork(vnet)
+            .withSubnet(subnetName)
+            .withPrivateIpAddressDynamic()
+            .withNewPublicIpAddress()
+            .withPublicIPAddressDeleteOptions(DeleteOptions.DETACH)
+            .attach()
+            .create();
+
+        nic.refresh();
+        nic.update()
+            .updateIPConfiguration("secondary")
+            .withNewPublicIpAddress()
+            .withPublicIPAddressDeleteOptions(DeleteOptions.DELETE)
```

Comment:
Assert different PublicIpAddress?

Also, seems to be missing the case where PublicIpAddress is not replaced, only DeleteOptions changing.